src/v/kafka/server/handlers/fetch.cc

```diff
@@ -201,6 +201,10 @@ static ss::future<read_result> do_read_from_ntp(
           ntp_config.cfg.isolation_level
           == model::isolation_level::read_committed) {
             ntp_config.cfg.max_offset = kafka_partition->last_stable_offset();
+            if (ntp_config.cfg.max_offset > model::offset{0}) {
```

Comment:
presumably this allows `0` to remain a readable offset. that shouldn't be a problem now because our groups always have a configuration batch in them at offset `0`, but we've discussed removing configuration batches. if we were to do that, would (1) this check need to be further refined and (2) if refined were needed, is there something that will help us remember that it is needed?

Reply:
This approach will work even without the configuration batch. When a transactional producer starts working with a partition it writes a fencing batch before its very first transaction so we may be sure that the zero-th batch is always there and doesn't contain data (or it contains data but it that case it isn't a part of a transaction).